packages/react-components/src/components/RichTextEditor/Plugins/CopyPastePlugin.tsx

```diff
@@ -50,12 +55,40 @@ const handleBeforePasteEvent = (
   }
 };
 
+/* @conditional-compile-remove(rich-text-editor-image-upload) */
+/**
+ * @internal
+ * Exported only for unit testing
+ */
+export const handleInlineImage = (
+  event: PluginEvent,
+  onUploadInlineImage?: (image: string, fileName: string) => void
+): void => {
+  if (event.eventType === PluginEventType.BeforePaste && event.pasteType === 'normal' && onUploadInlineImage) {
```

Comment:
Do we hae to handle these other types, such as AsImage?
![](image.png)

Reply:
@palatter In Rooster's demo app, those other types seem to only be used in a context menu where user can select how they want to paste the content. We don't have that context menu so the content will only be pasted in the default way, which is 'normal'.